src/execution/mod.rs

```diff
@@ -146,6 +146,20 @@ fn build_production_dataflow<A: Allocate>(
                 Step::Filter { predicate } => {
                     stream = stream.filter(move |item| filter(&predicate, item));
                 }
+                Step::FoldEpoch { builder, folder } => {
+                    stream = stream.map(extract_state_pair).aggregate(
```

Comment:
This comment isn't about this specific line of code, but is a general thing: Based on our discoveries as part of https://github.com/bytewax/bytewax/pull/96 , we probably won't have any operators that combine over epochs any more because epochs are going to be an implementation detail. So instead in the future this would become a `fold_window` operator. But! I don't think you should throw away this PR. We can move this over to use windowing stuff when it is done. I think Dan recorded a video of us discussing what it means to have "operational epochs" but I'm also happy to answer any questions.

Reply:
Yes I was talking with Dan about this yesterday. Let me know how you want to proceed, we can merge this PR, wait for the windowing one to be merged, and fix this in another PR, or we can wait for the windowing PR to be merged, fix this one and only after merge it.

Reply:
Sorry I didn't get back to you sooner. I already merged the windowing and source PR. So that will mean the tests need to be fixed up slightly. But I think that'll mean that our public API has stabilized and there shouldn't be any more changes at least!

Reply:
No problems, rebasing it on top of the new code made me understand some of the things going on in that PR, so I think that's good.